lib/is.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+// Ref; https://github.com/OpenObservability/OpenMetrics/blob/1dcecc569894abef6b67037b5e14f1813c7b84d7/metric_exposition_format.md
+const REGEX_NAME = /^[a-zA-Z_][a-zA-Z0-9_]*$/;
+const REGEX_LABEL = /^[a-zA-Z0-9_]*$/;
```

Comment:
hmm, I'm wondering if this may bite us at some point with existing metrics. I think that most if not all labels will comply but we may need to make some adjustments once we release to comply. We shall see.

Reply:
So, thing is: the Open Metric spec is not really a spec as is. They have intentions, but the spec is super fare from something one can sit down and just implement. The current regexes come from here which is when I started looking at this is the closest I came to a spec: https://github.com/OpenObservability/OpenMetrics/blob/1dcecc569894abef6b67037b5e14f1813c7b84d7/metric_exposition_format.md#notes-some-of-them-prometheus-specific

As I read them, no `-` char is valid.

Though; two days ago a PR with a proto definition (which they use to define the spec) occured:
https://github.com/OpenObservability/OpenMetrics/blob/9a6db2a94dcf8dd84dd08a221e1b323e2b279f08/proto/openmetrics_data_model.proto#L42

This one has a slightly different regex definition. A separate regex for label name is gone and the regex allow for `:`.

Though; no `-` it seems too.

Its a bit fluffy but if we encounter any `-` I kinda think that the usage of this is still at such a low level that its easier to handle it where we might encounter a `-` than adding a `-` in the validation and having to remove it later.

What do you think?

Reply:
Now using the new regex